router/router.go

```diff
@@ -73,6 +85,11 @@ func NewJsonDirectoryServer(schemaDirectory string, validator openrtb_ext.Bidder
 		data[bidder] = json.RawMessage(validator.Schema(bidderName))
 	}
 
+	//Add in any aliases
+	for aliasName, parentBidder := range getAliasBidders.getAliasBidderNames() {
```

Comment:
In this code, the `getAliasBidders` function is utilized within the `NewJsonDirectoryServer` function but is initialized outside of it. This essentially makes `getAliasBidders` a variable that can be accessed across the entire package.

Having `getAliasBidders` as a package variable offers certain advantages. 
- For instance, it allows us to easily mock the results of `getAliasBidderNames()` when testing. 
- Furthermore, it avoids the need to modify the callers of the `NewJsonDirectoryServer()` function to pass `getAliasBidders` as an argument.

However, this approach also introduces some additional code complexity. Specifically, it requires defining an interface and maintaining implementations that:
- Call `openrtb_ext.GetAliasBidderNames()` to obtain the actual `aliasToParent` mapping.
- Provide mock `aliasToParent` data for unit testing purposes.

Instead of this, how about making introducing`newJsonDirectoryServer()`. This new function takes aliasToParent as input. 


```
func NewJsonDirectoryServer(schemaDirectory string, validator openrtb_ext.BidderParamValidator, aliases map[string]string) httprouter.Handle {
	return newJsonDirectoryServer(schemaDirectory, validator, aliases, openrtb_ext.GetAliasBidderNames())
}


func newJsonDirectoryServer(schemaDirectory string, validator openrtb_ext.BidderParamValidator, aliases map[string]string, aliasToParent map[openrtb_ext.BidderName]openrtb_ext.BidderName) httprouter.Handle {
	// Slurp the files into memory first, since they're small and it minimizes request latency.
	files, err := os.ReadDir(schemaDirectory)
	if err != nil {
		glog.Fatalf("Failed to read directory %s: %v", schemaDirectory, err)
	}

	...


	//Add in any aliases
	for aliasName, parentBidder := range aliasToParent {
		data[string(aliasName)] = json.RawMessage(validator.Schema(parentBidder))
	}

	...

	return func(w http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
		w.Header().Add("Content-Type", "application/json")
		w.Write(response)
	}	

}

```
With above change, there's no longer a necessity to define an interface and its corresponding implementors. This simplifies the code setup and makes it cleaner. 

Additionally, we can directly pass mock aliasToParent data as an input parameter to newJsonDirectoryServer() without the need to declare a separate mock data provider.

Reply:
excellent suggestion, thanks! 
- https://github.com/prebid/prebid-server/pull/3055/commits/d5c207f18d210f765ed3301ddcc857da973a8b60